pkg/ingester/client/wrapped.go

```diff
@@ -0,0 +1,414 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package client
+
+import (
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+)
+
+const (
+	maxInPoolChunksSliceSize = 16_384
+)
+
+var (
+	chunkSlicesPool = sync.Pool{
+		New: func() interface{} { return &[]Chunk{} },
+	}
+)
+
+// IngesterQueryStreamClientWrappedReceiver extends the Ingester_QueryStreamClient interface
+// adding a wrapped response receiver method.
```

Comment:
What is the point of the wrapped response receiver? Why would I use it instead of the `Ingester_QueryStreamClient`?

Reply:
added extra comment @ 264d112a52876f4a2048fdcd3606efca28fb7f6e

hope it's clearer now.